hecuba_py/src/hfilter.py

```diff
@@ -0,0 +1,241 @@
+import regex
+import inspect
+import re
+from hecuba import config
+from IStorage import IStorage
+
+
+def func_to_str(func):
+    func_string = inspect.getsourcelines(func)[0][0]
+    start, end = func_string.find("lambda"), func_string.rfind(",")
+    func_string = func_string[start:end]
+    func_vars = func_string[7:func_string.find(':')].replace(" ", "").split(',')
+    clean_string = func_string[func_string.find(':') + 1:].replace("\\n", '').replace("'", '')
+    return func_vars, clean_string
+
+
+def substit_var(final_list, func_vars, dictv):
+    list_with_values = []
+    for elem in final_list:
+        if isinstance(elem, tuple) or isinstance(elem, set) or isinstance(elem, list):
+            list_with_values.append(elem)
+        elif (elem != 'in' and not isinstance(elem, int) and not regex.match('[^\s\w]', elem)) and (not elem.isdigit()):
+            if elem.find('.') > 0:
+                elem_var = elem[:elem.find('.')]
+                if elem_var not in func_vars:
+                    elemm = elem[elem.find('.'):]
+                    get_ele = dictv.get(str(elemm))
+                    if get_ele is None:
+                        list_with_values.append(elem)
+                    else:
+                        list_with_values.append(dictv.get(str(elem)))
+                else:
+                    list_with_values.append(elem[elem.find('.') + 1:])
+            else:
+                get_ele = dictv.get(str(elem))
+                if get_ele is None:
+                    list_with_values.append(elem)
+                else:
+                    list_with_values.append(dictv.get(str(elem)))
+        else:
+            list_with_values.append(elem)
+
+    return list_with_values
+
+
+def istype(var):
+    try:
```

Comment:
if the number has a dot '.' is a float, otherwise int right?
And if it has something not numerical is a str?